### ISSUE TITLE
v0.5.3 cannot revoke user-id

### ISSUE THREAD

**Issue by ZeeShen** [hashicorp/vault#1557]
My operation:

```
root@vault-stg1:~# vault -v
Vault v0.5.3
root@vault-stg1:~# vault write secret/test value=hello
Success! Data written to: secret/test

root@vault-stg1:~# vault write auth/app-id/map/app-id/foo value=root display_name=foo
Success! Data written to: auth/app-id/map/app-id/foo
root@vault-stg1:~# vault write auth/app-id/map/user-id/bar value=foo
Success! Data written to: auth/app-id/map/user-id/bar
```

After mapping, login & read secret is ok
```
root@vault-stg1:~# curl -d '{"app_id": "foo", "user_id":"bar"}' "http://localhost:8087/v1/auth/app-id/login"
{"lease_id":"","renewable":false,"lease_duration":0,"data":null,"warnings":null,"auth":{"client_token":"203edcfe-3a5a-17e8-a56f-ea42d6bd5e4b","accessor":"c95aec5d-4a48-97de-8c94-0231dfb46fef","policies":["root"],"metadata":{"app-id":"sha1:0beec7b5ea3f0fdbc95d0dd47f3c5bc275da8a33","user-id":"sha1:62cdb7020ff920e5aa642c3d4066950dd1f01f4d"},"lease_duration":0,"renewable":true}}

root@vault-stg1:~# curl -XGET -H "X-Vault-Token: 203edcfe-3a5a-17e8-a56f-ea42d6bd5e4b" "http://localhost:8087/v1/secret/test"
{"lease_id":"","renewable":false,"lease_duration":2592000,"data":{"value":"hello"},"warnings":null,"auth":null}
```

After revoke, still can login & read secret with either new or old token
```
root@vault-stg1:~# vault revoke auth/app-id/map/user-id/bar
Key revoked with ID 'auth/app-id/map/user-id/bar'.
root@vault-stg1:~# curl -XGET -H "X-Vault-Token: 203edcfe-3a5a-17e8-a56f-ea42d6bd5e4b" "http://localhost:8087/v1/secret/test"
{"lease_id":"","renewable":false,"lease_duration":2592000,"data":{"value":"hello"},"warnings":null,"auth":null}

root@vault-stg1:~# curl -d '{"app_id": "foo", "user_id":"bar"}' "http://localhost:8087/v1/auth/app-id/login"
{"lease_id":"","renewable":false,"lease_duration":0,"data":null,"warnings":null,"auth":{"client_token":"bdccad6a-0a69-5b73-360d-39f85a13f724","accessor":"49fcde9b-1bf7-a2b7-b8bf-7d28567082c3","policies":["root"],"metadata":{"app-id":"sha1:0beec7b5ea3f0fdbc95d0dd47f3c5bc275da8a33","user-id":"sha1:62cdb7020ff920e5aa642c3d4066950dd1f01f4d"},"lease_duration":0,"renewable":true}}

root@vault-stg1:~# curl -XGET -H "X-Vault-Token: bdccad6a-0a69-5b73-360d-39f85a13f724" "http://localhost:8087/v1/secret/test"
{"lease_id":"","renewable":false,"lease_duration":2592000,"data":{"value":"hello"},"warnings":null,"auth":null}
```

And I tried to delete mapping, cannot login  but still can read secret. 
```
root@vault-stg1:~# vault delete auth/app-id/map/user-id/bar
Success! Deleted 'auth/app-id/map/user-id/bar'

root@vault-stg1:~# curl -XGET -H "X-Vault-Token: 203edcfe-3a5a-17e8-a56f-ea42d6bd5e4b" "http://localhost:8087/v1/secret/test"
{"lease_id":"","renewable":false,"lease_duration":2592000,"data":{"value":"hello"},"warnings":null,"auth":null}
root@vault-stg1:~# curl -XGET -H "X-Vault-Token: bdccad6a-0a69-5b73-360d-39f85a13f724" "http://localhost:8087/v1/secret/test"
{"lease_id":"","renewable":false,"lease_duration":2592000,"data":{"value":"hello"},"warnings":null,"auth":null}
```

I saw you closed [this issue](https://github.com/hashicorp/vault/issues/191) at 2015 and v0.5.3 released this year. So could you please help to check that this revoke mapping is working or not? And I think that after revoke, neither login nor read secret can be done, is that right?

**Comment by jefferai** [comment]
@ZeeShen Revocation is done at a token or lease level -- not at an app-id level. Deleting an app-id mapping doesn't revoke issued tokens, and running a revocation request against that endpoint does not take any action.

If you want to revoke all tokens generated by app-id, you could do a `vault revoke -prefix` operation against `auth/app-id`, which should do what you want.

**Comment by ZeeShen** [comment]
Thanks, @jefferai. 
But what I want is to disable certain `user-id` from reading secrets(e.g., when stop a machine that related to this user-id). Meanwhile, I want to keep related `app-id`s.
I tried `vault revoke auth/app-id/map/user-id/bar` because I saw in this [issue](https://github.com/hashicorp/vault/issues/191) @sethvargo  said that:

> sethvargo commented on May 13, 2015

> Hi @jsok,
> 
> I believe you can use the revoke command here:
> 
> vault revoke auth/app-id/map/user-id/bar
> Or, if you wanted to disable auth for that app entirely:
> 
> vault revoke -prefix auth/app-id
> Does that answer your question?

So if revoting a user-id didn't work, what should I do to disable a user-id and revoke all its token?
